prov/util/src/util_av.c

```diff
@@ -398,6 +399,7 @@ static void util_av_close(struct util_av *av)
 {
 	HASH_CLEAR(hh, av->hash);
 	ofi_bufpool_destroy(av->av_entry_pool);
+	fi_close(&av->coll_mc->mc_fid.fid);
```

Comment:
The util_av code is common and may not use collectives.  This change doesn't look right.

Reply:
The code is a bit strange; the av->coll_mc is allocated by ofi_av_set in the util_coll.c. I think it's an optimization to avoid recreating it and is used for FI_ADDR_NOTAVAIL.

I changed the code to free everything there without relying on fi_close as it is specific to av and is a special case.